src/module.c

```diff
@@ -1039,9 +1054,16 @@ JL_DLLEXPORT void jl_disable_binding(jl_globalref_t *gr)
 
     JL_LOCK(&world_counter_lock);
     jl_task_t *ct = jl_current_task;
+    size_t last_world = ct->world_age;
     size_t new_max_world = jl_atomic_load_acquire(&jl_world_counter);
-    // TODO: Trigger invalidation here
-    (void)ct;
+    JL_TRY {
+        ct->world_age = jl_typeinf_world;
+        jl_invalidate_binding_refs(gr, new_max_world);
+    } JL_CATCH {
+        JL_UNLOCK(&world_counter_lock);
```

Comment:
You are forbidden from unlocking a lock that you didn't acquire in the same scope
```suggestion
```